src/Hooks.ts

```diff
@@ -16,9 +16,14 @@ export function useNavigation<S>(): NavigationScreenProp<S & NavigationRoute> {
 }
 
 export function useNavigationParam<T extends keyof NavigationParams>(
-  paramName: T
+  paramName: T,
+  fallbackValue?
 ) {
-  return useNavigation().getParam(paramName);
+  const { getParam, setParams } = useNavigation()
+  return [
+    getParam(paramName, fallbackValue),
+    (newValue: NonNullable<NavigationParams[T]>) => setParams({ [paramName]: newValue }),
```

Comment:
Why `NonNullable`? `null` should be a valid value

Reply:
You're completely right - this is the first time I do something with typescript outside of a tutorial 😓